openedx/core/djangoapps/content/search/documents.py

```diff
@@ -376,11 +391,15 @@ def searchable_doc_for_library_block(xblock_metadata: lib_api.LibraryXBlockMetad
     library_name = lib_api.get_library(xblock_metadata.usage_key.context_key).title
     block = xblock_api.load_block(xblock_metadata.usage_key, user=None)
 
+    publish_status = PublishStatus.published
     try:
         block_published = xblock_api.load_block(xblock_metadata.usage_key, user=None, version=LatestVersion.PUBLISHED)
+        if xblock_metadata.last_published and xblock_metadata.last_published < xblock_metadata.modified:
+            publish_status = PublishStatus.modified
     except NotFound:
         # Never published
         block_published = None
+        publish_status = PublishStatus.never
```

Comment:
Hmm.. technically this status can be determined using filters on the existing xblock document metadata fields, so we don't strictly _need_ a field `publish_status` field in the search index.  e.g to filter out unpublished items when sorting by "Recently Published", we add a [`last_published IS NOT NULL`](https://github.com/openedx/frontend-app-authoring/blob/master/src/search-manager/SearchManager.ts#L136) filter.

But I'm not sure what our stance is on having too much logic living in the frontend.. @bradenmacdonald what do you think?

Reply:
@pomegranited I agree, it seems odd to add this if it's derivable. But that's the catch - I had originally started working on this in https://github.com/openedx/frontend-app-authoring/pull/1406 and I found this was necessary:

> [we will need] add a filterable "publish_status" field on the backend Meilisearch index which is set to "published", "modified", or "never". It seems that this is necessary because Meilisearch can filter on things like `modified > 23849345` but NOT on things like `modified > last_published`

If you can find a way to implement this without changing the backend/index, I'd be much happier but I didn't think it was possible.

Reply:
Oh! Got it :) Thanks for the explanation.